requre/online_replacing.py

```diff
@@ -292,3 +305,117 @@ def _internal(*args, **kwargs):
         return _internal
 
     return decorator_cover
+
+
+def record(
+    what: str, storage_file: Optional[str] = None,
+):
+    """
+    Decorator which can be used to store calls of the function and
+    and replay responses on the next run.
+
+    - Uses pickle for storing the responses => the objects need to be (de)serializable.
+    - Matching is based on the all arguments.
+
+    :param what: Name of the callable that will be run for the first time
+                    and the response will be used when running for the next time.
+    :param storage_file: file for reading and writing data in storage_object
+    """
+
+    def _record_inner(func):
+        return replace_module_match(
+            what=what,
+            decorate=ObjectStorage.decorator_all_keys,
+            storage_file=storage_file,
+        )(func)
+
+    return _record_inner
+
+
+def record_requests(
+    _func=None, response_headers_to_drop: Optional[List[str]] = None, storage_file=None
+):
+    """
+    Decorator which can be used to store all requests to a file
+    and replay responses on the next run.
+
+    - The matching is based on `url`.
+    - Removes tokens from the url when saving if needed.
+
+    Can be used with or without parenthesis.
+
+    :param _func: can be used to decorate function (with, or without parenthesis).
+    :param response_headers_to_drop: list of header names we don't want to save with response
+                                        (Will be replaced to `None`.)
+    :param storage_file: file for reading and writing data in storage_object
+    """
+
+    response_headers_to_drop = response_headers_to_drop or []
+
+    def decorator_cover(func):
+        return replace_module_match(
+            what="requests.sessions.Session.request",
+            decorate=RequestResponseHandling.decorator(
+                item_list=["method", "url", "data"],
```

Comment:
I know that I've also added `"data"` but data may contains sensitive info, so maybe `"method", "url"` should be enough and secure.